packages/perspective-jupyterlab/src/ts/renderer.ts

```diff
@@ -82,19 +80,14 @@ export class PerspectiveDocumentWidget extends DocumentWidget<PerspectiveWidget>
                 // don't handle other mimetypes for now
                 throw "Not handled";
             }
-            perspective
-                .worker()
-                .table(data)
-                .then((table: any) => {
-                    this._table = table;
-                    if (this._psp.viewer.table === undefined) {
-                        // construct new table
-                        this._psp.viewer.load(this._table);
-                    } else {
-                        // replace existing table for whatever reason
-                        this._psp.replace(this._table);
-                    }
-                });
+
+            if (this._psp.viewer.table === undefined) {
+                // construct new table
+                this._psp.viewer.load(perspective.worker().table(data));
```

Comment:
This will create a new WebWorker for each widget - the worker should be a property on the DocumentWidget.